### ISSUE TITLE
Doesn't seem to work with `...attributes`

### ISSUE THREAD

**Issue by knownasilya** [tildeio/ember-element-helper#6]
```hbs
{{#let (element this.tagName) as |Tag|}}
  <Tag
    {{did-insert this.setupRoot}}
    ...attributes
  ></Tag>
{{/let}}
```

Seems to just not work. No errors or anything. In this case `tagName` is a 'span'.
Full example is here https://github.com/knownasilya/ember-shadow-dom/pull/1/commits/429b88121d5717b20dcfbda7821030459516c2d7

**Comment by knownasilya** [comment]
The `...attributes` and element modifier are actually not the cause. It seems like it just doesn't work within the octane blueprint (addon in my case).

**Comment by knownasilya** [comment]
cc @chancancode

**Comment by chancancode** [comment]
I tried the following in a fresh octane blueprint app (3.10.0-canary+707a938e), and it seems to work:

```hbs
{{#let (element 'a') as |Tag|}}
  <Tag href='/'>ohai</Tag>
{{/let}}
```

**Comment by knownasilya** [comment]
@chancancode I think the actual issue is the element modifier
```hbs
<Tag
    {{did-insert this.setupRoot}}
  ></Tag>
```

**Comment by jonnygreen** [comment]
I'm seeing this exact issue with Ember 3.5.

See this code sandbox https://codesandbox.io/s/ember-element-helper-test-35-l5hg6.

Tag with `...attributes` does not render.

It's currently affecting `ember-animated` 0.7.0 (see https://github.com/ember-animation/ember-animated/issues/146)

**Comment by ryanto** [comment]
Heya!

It looks like the presence of `...attributes` with `ember-element-helper` break when using Ember `3.5.0 - ~3.7.1`. This commit, which was released in 3.7.2 fixes the issue: https://github.com/emberjs/ember.js/commit/c801853e188ab179d35731f6d3e8cd1e9b860c14

Here's a minimal reproduction of the issue: https://github.com/ryanto/element-helper-reproduction Bump the `ember-source` version to 3.7.2 to see the expected behavior.

Let me know if there's any more info I can provide!

**Comment by Panman82** [comment]
So just to clarify, is this no longer an issue when used with ember 3.7.2+? And whatever fixed it in 3.7.2 cannot be back-ported to earlier ember versions, or handled by this addon itself? (FWIW, I'm shooting for backwards compatibility to 3.4 in another addon.)

**Comment by ryanto** [comment]
@Panman82 Since the fix was an update to the version glimmer used by ember-source, that glimmer update would have to be backported to the older Ember versions iirc.

**Comment by Panman82** [comment]
@ryanto sure, understandable that it's unlikely that such things would be backported, since we're so far away from 3.7 anyway. Should this addon have compatibility set to 3.7.2+ and this issue closed?? @chancancode ?

**Comment by chancancode** [comment]
There are two issues here, both of which are Ember.js issues/limitations and unrelated to this addon, because they apply to contextual components in general, not just the ones generated using the `(element)` helper provided here.

1. `...attributes` usage
2. modifiers usage

The fix for `...attributes` on contextual components has been fixed on Ember.js and was backported to all LTS versions supported by this addon, including LTS 3.4, so it shouldn't be a problem anymore. At this point, if you are on an old intermediate, non LTS version, then unfortunately you are probably on your own, which is the point of the LTS policy.

As for modifiers, support for that feature was introduced in [RFC #435](https://github.com/emberjs/rfcs/blob/master/text/0435-modifier-splattributes.md) and is available in 3.11+. Unfortunately, there is no polyfill available as the feature was implemented pretty deep inside glimmer-vm. Again, it is a limitation for contextual components in general, not just the ones generated by this addon.

I added test for both and documented the second limitation in #23. Once that lands, this issue should be closed.
